eh_personality.c

```diff
@@ -760,3 +761,10 @@ void objc_exception_rethrow(struct _Unwind_Exception *e)
 	_Unwind_Resume_or_Rethrow(e);
 	abort();
 }
+
+objc_uncaught_exception_handler objc_setUncaughtExceptionHandler(objc_uncaught_exception_handler handler)
+{
+	objc_uncaught_exception_handler previousHandler = _objc_unexpected_exception;
```

Comment:
This should be an atomic exchange.

Reply:
Sure, would you use `atomic_exchange()` and declare `_objc_unexpected_exception` as volatile, or is there another way without needing volatile?

Btw. the Apple runtime doesn’t do an atomic exchange:
https://github.com/opensource-apple/objc4/blob/cd5e62a5597ea7a31dccef089317abb3a661c154/runtime/objc-exception.mm#L455

Reply:
Sorry which macros exactly? Do you mean `objc_atomicCompareAndSwapPtr()`?

Reply:
Sorry I’m not so familiar with builtins. Would the following make sense at all?
```
return __atomic_exchange(&_objc_unexpected_exception, handler, __ATOMIC_RELAXED);
```

Reply:
I ended up using the following, I hope that makes sense?
```c
__atomic_exchange_n(&_objc_unexpected_exception, handler, __ATOMIC_SEQ_CST);
```

Reply:
Looks good.  We probably can get away with a slightly weaker memory order, but if anyone calls this enough times for it to matter for performance then they deserve for it to be slow.

Reply:
Heh sounds good. I just copied this from somewhere... 🙈  Just let me know if I should change it in any way.

The test is also excluded on ARM now so CI is green.